erdkotlin/src/main/java/com/elrond/erdkotlin/domain/wallet/Wallet.kt

```diff
@@ -0,0 +1,185 @@
+package com.elrond.erdkotlin.domain.wallet
+
+import com.elrond.erdkotlin.Exceptions
+import org.bitcoinj.crypto.MnemonicCode
+import org.bitcoinj.crypto.MnemonicException.MnemonicLengthException
+import org.bouncycastle.crypto.digests.SHA512Digest
+import org.bouncycastle.crypto.generators.PKCS5S2ParametersGenerator
+import org.bouncycastle.crypto.macs.HMac
+import org.bouncycastle.crypto.params.Ed25519PrivateKeyParameters
+import org.bouncycastle.crypto.params.KeyParameter
+import org.bouncycastle.crypto.signers.Ed25519Signer
+import org.bouncycastle.util.encoders.Hex
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.nio.charset.StandardCharsets
+import java.security.SecureRandom
+
+
+class Wallet(
+    private val publicKey: ByteArray,
+    private val privateKey: ByteArray,
+) {
+
+    val publicKeyHex: String = String(Hex.encode(publicKey))
+    val privateKeyHex: String = String(Hex.encode(privateKey))
+
+    fun sign(data: String): String = sign(data.toByteArray(StandardCharsets.UTF_8))
+
+    fun sign(data: ByteArray): String {
+        val signer: Ed25519Signer = createEd25519Signer().apply {
+            update(data, 0, data.size)
+        }
+        val signature = signer.generateSignature()
+        val hex = Hex.encode(signature)
+        return String(hex)
+    }
+
+    private fun createEd25519Signer(): Ed25519Signer {
+        val parameters = Ed25519PrivateKeyParameters(this.privateKey, 0)
+        return Ed25519Signer().apply {
+            init(true, parameters)
+        }
+    }
+
+    companion object {
+        private const val DEFAULT_ENTROPY_BITS = 256 // this leads to 24-words mnemonics
+        private const val BIP39_SALT_MODIFIER = "mnemonic"
+        private const val BIP39_PBKDF2_ROUNDS = 2048
+        private const val BIP32_SEED_MODIFIER = "ed25519 seed"
+        private const val HARDENED_OFFSET: Long = -0x80000000
```

Comment:
Depends on how it is used, hardened paths are 0x800000XX, so it should be positive.

Reply:
Fixed.
Please look at the update